bundles/mapping/camera-controls-3d/tool/CameraControls3dTool.js

```diff
@@ -0,0 +1,72 @@
+Oskari.clazz.define('Oskari.mapping.cameracontrols3d.CameraControls3dTool',
+    function () {
+    }, {
+        index: 3,
+        allowedLocations: ['top left', 'top right', 'bottom left', 'bottom right'],
+        lefthanded: 'top left',
+        righthanded: 'top right',
+        allowedSiblings: [
+            'Oskari.mapframework.bundle.featuredata2.plugin.FeaturedataPlugin',
+            'Oskari.mapframework.bundle.mapmodule.plugin.MyLocationPlugin',
+            'Oskari.mapframework.bundle.mapmodule.plugin.Portti2Zoombar',
+            'Oskari.mapframework.bundle.mapmodule.plugin.PanButtons'
+        ],
+
+        groupedSiblings: true,
+
+        /**
+    * Get tool object.
+    * @method getTool
+    *
+    * @returns {Object} tool description
+    */
+        getTool: function () {
+            return {
+                id: 'Oskari.mapping.cameracontrols3d.CameraControls3dPlugin',
+                title: 'CameraControls3d',
+                config: {}
+            };
+        },
+        /**
+    * Get values.
+    * @method getValues
+    * @public
+    *
+    * @returns {Object} tool value object
+    */
+        getValues: function () {
+            var me = this;
+
+            if (me.state.enabled) {
+                return {
+                    configuration: {
+                        mapfull: {
```

Comment:
This is not part of mapfull plugins but a bundle called "camera-controls-3d". So getValues() should return something different. The featuredata bundle works in the same way so you could look for an example here https://github.com/oskariorg/oskari-frontend/blob/develop/bundles/framework/publisher2/tools/FeaturedataTool.js#L62

Reply:
You will also need to whitelist the bundle for publishing in oskari-server: https://github.com/oskariorg/oskari-server/blob/master/control-base/src/main/java/fi/nls/oskari/control/view/AppSetupHandler.java#L149

Reply:
Thank you for the tips. Camera controls are now handled as a bundle. Also https://github.com/oskariorg/oskari-server/pull/497 is created to whitelist the bundle.